driver-core/src/main/com/mongodb/client/model/expressions/ExpressionCodec.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import com.mongodb.annotations.Immutable;
+import org.bson.BsonReader;
+import org.bson.BsonValue;
+import org.bson.BsonWriter;
+import org.bson.codecs.BsonValueCodecProvider;
+import org.bson.codecs.Codec;
+import org.bson.codecs.DecoderContext;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+
+import static org.bson.codecs.configuration.CodecRegistries.fromProviders;
+
+@Immutable
+public final class ExpressionCodec implements Codec<Expression> {
```

Comment:
This should be:

     class ExpressionCodec<T extends Expression> implements Codec<T>

and the constructor should have a `Class<T> clazz` parameter, which can then be returned from `getEncoderClass`

Reply:
Or if we can only encode MqlExpression instances, then, it should be:

    class ExpressionCodec<T extends MqlExpression> implements Codec<T>

and we should have an `MqlExpresionCodecProvider`

Reply:
How would `ExpressionCodec<T extends Expression>` come into play? That is, what would have to come after `BooleanExpression b = ???` for this to be necessary?

Reply:
I'm wrong.  This can just be:

    class MqlExpressionCodec implements Codec<MqlExpression>